### PR TITLE
Intel 18 container

### PR DESCRIPTION
**Pull request by KaiSzuttor**

### CODE REVIEW

File: docker/intel/Dockerfile-18 [hunk 1]
```diff
@@ -30,33 +30,23 @@ COPY intel.cfg /tmp/
 RUN cd /tmp \
  && mkdir intel \
  && cd intel \
- && curl -sL http://registrationcenter-download.intel.com/akdlm/irc_nas/tec/11541/parallel_studio_xe_2017_update4_composer_edition.tgz | tar --strip-components=1 -xz \
+ && curl -sL http://registrationcenter-download.intel.com/akdlm/irc_nas/tec/13717/parallel_studio_xe_2018_update4_cluster_edition_online.tgz | tar --strip-components=1 -xz \
```
Comment (mkuron) [inline docker/intel/Dockerfile-18]: Please use composer edition, it is a smaller download.

File: docker/intel/Dockerfile-18 [hunk 2]
```diff
@@ -30,33 +30,23 @@ COPY intel.cfg /tmp/
...
-RUN cd /tmp \
- && mkdir intel-mpi \
- && cd intel-mpi \
- && curl -sL http://registrationcenter-download.intel.com/akdlm/irc_nas/tec/12748/l_mpi_2018.2.199.tgz | tar --strip-components=1 -xz \
- && ./install.sh --silent /tmp/intel-mpi.cfg \
- && ln -s /opt/intel-mpi/compilers_and_libraries_2018.2.199/linux/mpi/intel64/bin/mpicxx /opt/intel-mpi/compilers_and_libraries_2018.2.199/linux/mpi/intel64/bin/mpic++ \
- && cd \
- && rm -r /tmp/intel-mpi
-
-ENV PATH="/opt/intel-mpi/compilers_and_libraries_2018.2.199/linux/mpi/intel64/bin:${PATH}" I_MPI_ROOT=/opt/intel-mpi/compilers_and_libraries_2018.2.199/linux/mpi
```
Comment (mkuron) [inline docker/intel/Dockerfile-18]: and keep Intel MPI

File: docker/intel/intel.cfg [hunk 3]
```diff
@@ -4,7 +4,6 @@ CONTINUE_WITH_OPTIONAL_ERROR=yes
 PSET_INSTALL_DIR=/opt/intel
 CONTINUE_WITH_INSTALLDIR_OVERWRITE=yes
 PSET_MODE=install
-PHONEHOME_SEND_USAGE_DATA=no
 SIGNING_ENABLED=yes
-COMPONENTS=intel-icc-l-ps-ss-bec-wrapper__x86_64;intel-icc-l-all__x86_64;intel-compxe__noarch;intel-ccompxe__noarch;intel-psxe-licensing__noarch;intel-psxe-common__noarch
+COMPONENTS=DEFAULTS
```
Comment (mkuron) [inline docker/intel/intel.cfg]: don't do that, it produces a huge images with stuff we don't need (GUI applications, Fortran compiler, etc.)

File: docker/intel/Dockerfile-18 (hunk 1 above)
Comment (KaiSzuttor) [reply docker/intel/Dockerfile-18]: That only happens once, so I'll keep it for now

File: docker/intel/Dockerfile-18 [hunk 4]
```diff
@@ -30,33 +30,25 @@ COPY intel.cfg /tmp/
...
  && mkdir intel \
  && cd intel \
- && curl -sL http://registrationcenter-download.intel.com/akdlm/irc_nas/tec/11541/parallel_studio_xe_2017_update4_composer_edition.tgz | tar --strip-components=1 -xz \
+ && curl -sL http://registrationcenter-download.intel.com/akdlm/irc_nas/tec/13717/parallel_studio_xe_2018_update4_cluster_edition_online.tgz | tar --strip-components=1 -xz \
  && USER=root ./install.sh --silent /tmp/intel.cfg \
  && cd \
- && rm -r /tmp/intel
+ && rm -r /tmp/intel \
+ && rm -r /opt/intel/inspector* /opt/intel/vtune_amplifier* /opt/intel/ism /opt/intel/clck /opt/intel/advisor /opt/intel/documentation* \
+ /opt/intel/debugger* /opt/intel/itac /opt/intel/compilers_and_libraries_2018.5.274/linux/daal /opt/intel/compilers_and_libraries_2018.5.274/linux/ipp /tmp/parallel_studio_xe_2018_update4_cluster_edition
```
Comment (mkuron) [inline docker/intel/Dockerfile-18]: This is ridiculous. Just properly select the components you install.

Comment (KaiSzuttor) [reply docker/intel/Dockerfile-18]: If you tell me the names for intel18, I don't have time to look them up.

**Comment by mkuron** [comment]
To get the names of the packages, just install Intel Parallel Studio from the GUI, deselecting everything but the C++ compiler, and then grep for `PRODUCTS_ABBR_TO_INSTALL` in /tmp/intel.pset.*.log. I assume you can even remove everything _conda_-, _tbb_-, _doc_- and _32bit_-related from these lists.

http://registrationcenter-download.intel.com/akdlm/irc_nas/tec/14832/parallel_studio_xe_2019_update1_composer_edition.tgz

> ;intel-conda-index-tool__x86_64;intel-comp__x86_64;intel-comp-32bit__x86_64;intel-comp-doc__noarch;intel-comp-l-all-common__noarch;intel-comp-l-all-vars__noarch;intel-comp-nomcu-vars__noarch;intel-comp-ps-32bit__x86_64;intel-comp-ps__x86_64;intel-comp-ps-ss__x86_64;intel-comp-ps-ss-bec__x86_64;intel-comp-ps-ss-bec-32bit__x86_64;intel-openmp__x86_64;intel-openmp-32bit__x86_64;intel-openmp-common__noarch;intel-openmp-common-icc__noarch;intel-tbb-libs-32bit__x86_64;intel-tbb-libs__x86_64;intel-idesupport-icc-common-ps__noarch;intel-conda-icc_rt-linux-64-shadow-package__x86_64;intel-icc__x86_64;intel-c-comp-common__noarch;intel-icc-common__noarch;intel-icc-common-ps__noarch;intel-icc-common-ps-ss-bec__noarch;intel-icc-doc__noarch;intel-icc-doc-ps__noarch;intel-icc-ps__x86_64;intel-icc-ps-ss-bec__x86_64;intel-tbb-devel__x86_64;intel-tbb-common__noarch;intel-tbb-doc__noarch;intel-conda-tbb-linux-64-shadow-package__x86_64;intel-conda-tbb-devel-linux-64-shadow-package__x86_64;intel-ism__noarch;intel-compxe__noarch;intel-psxe-common__noarch;intel-psxe-doc__noarch;intel-psxe-common-doc__noarch;intel-compxe-doc__noarch;intel-psxe-licensing__noarch;intel-psxe-licensing-doc__noarch;intel-icsxe-pset

http://registrationcenter-download.intel.com/akdlm/irc_nas/tec/14879/l_mpi_2019.1.144.tgz

> DEFAULTS

http://registrationcenter-download.intel.com/akdlm/irc_nas/tec/13722/parallel_studio_xe_2018_update4_composer_edition.tgz

> ;intel-comp__x86_64;intel-comp-doc__noarch;intel-comp-l-all-common__noarch;intel-comp-l-all-vars__noarch;intel-comp-nomcu-vars__noarch;intel-comp-ps__x86_64;intel-comp-ps-ss__x86_64;intel-comp-ps-ss-bec__x86_64;intel-openmp__x86_64;intel-openmp-common__noarch;intel-openmp-common-icc__noarch;intel-tbb-libs__x86_64;intel-idesupport-icc-common-ps__noarch;intel-icc__x86_64;intel-c-comp-common__noarch;intel-icc-common__noarch;intel-icc-common-ps__noarch;intel-icc-common-ps-ss-bec__noarch;intel-icc-doc__noarch;intel-icc-doc-ps__noarch;intel-icc-ps__x86_64;intel-icc-ps-ss__x86_64;intel-icc-ps-ss-bec__x86_64;intel-tbb-devel__x86_64;intel-tbb-common__noarch;intel-tbb-doc__noarch;intel-ism__noarch;intel-compxe__noarch;intel-psxe-common__noarch;intel-psxe-doc__noarch;intel-psxe-common-doc__noarch;intel-compxe-doc__noarch;intel-psxe-licensing__noarch;intel-psxe-licensing-doc__noarch;intel-icsxe-pset

http://registrationcenter-download.intel.com/akdlm/irc_nas/tec/13741/l_mpi_2018.4.274.tgz

> DEFAULTS

**Comment by KaiSzuttor** [comment]
sorry @mkuron, I was faster

**Comment by KaiSzuttor** [comment]
offline approved by @mkuron

**Pull request merged**